### PR TITLE
Integration Tests - authorized nodes, build service and CLI options

### PR DESCRIPTION
**Pull request by karolh2000**
- New and upgraded integration tests related to "authorize" nodes.
- Tests refactoring, the build service test split into two tests: maven and docker.
- The scenarios are updated to use bootstrap and non-bootstrap scenarios including "authorized" nodes.
- Docker/maven covered in the context of two pyrsia nodes (local and authorized)

### CODE REVIEW

**Comment by karolh2000** [comment]
The repo/branch used for this IT is based on @erwin1's repo/branch:

  git_repo="https://github.com/erwin1/pyrsia.git"
  git_branch="1083-add-authorized-node"

It will be replaced by the main pyrsia branch when https://github.com/pyrsia/pyrsia/pull/1102 is merged

**Comment by karolh2000** [comment]
Blocker for this PR - https://github.com/pyrsia/pyrsia/issues/1135

**Comment by karolh2000** [comment]
The PR also contains changes related to the nodes and CLI updates.

File: readme.md [hunk 1]
```diff
@@ -50,11 +50,16 @@ REPO_DIR=<path to your integration tests repo> $REPO_DIR/bats/run_tests.sh
...
    - Test the build service, MAVEN (build, inspect-log).
-     - NOTE: This test is partly disabled because of <https://github.com/pyrsia/pyrsia/issues/1032>
    - Test the build service, DOCKER (build docker image, inspect-log)
 3) Pyrsia P2P service, client side (project)
    - TBD
+
+## Clean up tests environment
+
+The docker containers created by the tests framework are removed when CLEAN_UP_TEST_ENVIRONMENT=true (default).
+The docker images have to be removed manually. The Pyrsia integration tests also use the temp directory `/tmp/pyrsia_tests`
```
Comment (dasmanas) [inline readme.md:65]: **Suggestion** 
docker images can be cleaned by using `docker compose down --rmi local`.

**Review by dasmanas — APPROVED** [review]

File: .github/workflows/run-bats-tests.yml [hunk 2]
```diff
@@ -4,9 +4,10 @@ on:
   pull_request:
     types:
       - opened
+      - reopened
```
[the remark this answers is not in the document]
Comment (karolh2000) [reply .github/workflows/run-bats-tests.yml:7]: I don't think it is I have to investigate what event covers it.

File: readme.md (hunk 1 above)
Comment (karolh2000) [reply readme.md]: The images are actually removed, readme updated.

**Pull request merged**